flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -1124,20 +1124,21 @@ public static void main(final String[] args) {
         final List<CustomCommandLine> customCommandLines =
                 loadCustomCommandLines(configuration, configurationDirectory);
 
+        int retCode = -1;
         try {
             final CliFrontend cli = new CliFrontend(configuration, customCommandLines);
 
             SecurityUtils.install(new SecurityConfiguration(cli.configuration));
-            int retCode =
+            retCode =
                     SecurityUtils.getInstalledContext().runSecured(() -> cli.parseAndRun(args));
-            System.exit(retCode);
         } catch (Throwable t) {
             final Throwable strippedThrowable =
                     ExceptionUtils.stripException(t, UndeclaredThrowableException.class);
             LOG.error("Fatal error while running command line interface.", strippedThrowable);
             strippedThrowable.printStackTrace();
             System.exit(31);
```

Comment:
retCode = 31, and call `System#exit` method in finally block.

Reply:
Done.